platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/isolated/FetchAllIdeaProjects.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.configurationcache.isolated;
+
+import org.gradle.tooling.BuildAction;
+import org.gradle.tooling.BuildController;
+import org.gradle.tooling.model.gradle.GradleBuild;
+import org.gradle.tooling.model.idea.IdeaProject;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class FetchAllIdeaProjects implements BuildAction<FetchAllIdeaProjects.Result> {
+
+    @Override
+    public Result execute(BuildController controller) {
+        Result result = new Result();
+
+        GradleBuild buildModel = controller.getBuildModel();
+        result.rootBuild = buildModel;
+        result.rootIdeaProject = controller.getModel(IdeaProject.class);
+
+        collectAllNestedBuilds(buildModel, controller, result);
+
+        result.allIdeaProjects.add(result.rootIdeaProject);
+        LinkedHashMap<GradleBuild, IdeaProject> includedWithoutRoot = new LinkedHashMap<>(result.includedBuildIdeaProjects);
```

Comment:
💭 Is it necessary to wrap `result.includedBuildIdeaProjects` in a new `LinkedHashMap` ? Can't we use it as is?
I mean, I see the types, but I guess that making `includedWithoutRoot` as a `Map` would work here.

Reply:
@alllex Oh, never mind, I missed that we are do some mutation below at line 43. So we need some defensive copy to prevent `result.includedBuildIdeaProjects` be changed as well.